src/main/java/org/jenkinsci/plugins/workflow/support/steps/stash/StashStep.java

```diff
@@ -109,10 +111,10 @@ public boolean isAllowEmpty() {
             this.step = step;
         }
 
-        @Override protected Void run() throws Exception {
+        @Override protected List<String> run() throws Exception {
             StashManager.stash(getContext().get(Run.class), step.name, getContext().get(FilePath.class), getContext().get(TaskListener.class), step.includes, step.excludes,
                     step.useDefaultExcludes, step.allowEmpty);
-            return null;
+            return StashManager.stashFileList(getContext().get(Run.class), step.name);
```

Comment:
This seems inefficient and silly. Just make `stash` and `unstash` return the `List<String>` as they are processed, so you do not need to process the whole tarball twice.

Reply:
So the reason for this approach is that when we changed the signature for `StashManager.stash` and `StashManager.unstash` and released `workflow-api`, [users started seeing hangs with `stash`](https://issues.jenkins-ci.org/browse/JENKINS-40912?focusedCommentId=317875&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-317875) with the new `workflow-api` version. I never managed to reproduce the hang in tests, but I assumed it was due to the signature change, so this time I tried to avoid needing a signature change.